core-bundle/src/DependencyInjection/Attribute/AsContentElement.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\DependencyInjection\Attribute;
+
+/**
+ * Service tag to autoconfigure content elements.
+ */
+#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::IS_REPEATABLE)]
+class AsContentElement
+{
+    public array $attributes;
+
+    public function __construct(string $type = null, string $category, string $template = null, string $method = null, string $renderer = null, array ...$attributes)
```

Comment:
<img width="460" alt="" src="https://user-images.githubusercontent.com/1192057/141334962-c3fd467b-125e-40cb-ae66-b37d2d178b24.png">

Reply:
unfortunately, this is correct. The type is automatically inferred from the class name, but the category must be set. If one enters all arguments in the list, the type should be first, as it was with the annotation. But one can use names arguments to only set the category, exactly like now in annotations.